File: Checks/PWR056/example-scalar.c
// PWR056: consider applying offloading parallelism to scalar reduction loop

// NOT-PWR003: OpenMP/OpenACC offloading pragmas aren't considered `pure`
double example(double *A, int n) {
  double sum = 0.0;

  for (int i = 0; i < n; ++i) {
    sum += A[i];
  }

  return sum;
}

File: Checks/PWR054/example-scalar.f90
! PWR054: consider applying vectorization to scalar reduction loop

pure function example(A) result(sum)
  use iso_fortran_env, only: real32
  implicit none
  real(kind=real32), intent(in) :: A(:)
  real(kind=real32) :: sum
  integer :: i

  sum = 0.0
  do i = 1, size(A, 1)
    sum = sum + A(i)
  end do
end function example
